grails-app/services/com/unifina/service/BalanceService.groovy

```diff
@@ -35,24 +40,32 @@ class BalanceService {
 	}
 }
 
-interface Web3Balance {
-	BigInteger checkBalance(String address) throws InterruptedException, ExecutionException, MessageDecodingException
+abstract class Web3Balance {
+	abstract BigInteger getDatacoinBalance(String address) throws InterruptedException, ExecutionException, MessageDecodingException
+
+	Map<String, BigInteger> getDatacoinBalances(SecUser user) throws InterruptedException, ExecutionException, MessageDecodingException  {
+		def keys =
+			IntegrationKey.findAllByUserAndService(user, IntegrationKey.Service.ETHEREUM) +
+				IntegrationKey.findAllByUserAndService(user, IntegrationKey.Service.ETHEREUM_ID)
+		Map<String, BigInteger> rslt = new LinkedHashMap<String, BigInteger>();
+		for(IntegrationKey ik : keys){
+			rslt.put(ik.idInService, getDatacoinBalance(ik.idInService))
```

Comment:
Is there any easy way to parallelize the requests? Yeah, in JS this would be trivial, but not so simple in Java...